### ISSUE TITLE
[Feature request]: Support partial commit hashes for `update --commit` to downgrade packages

### ISSUE THREAD

**Issue by nekohayo** [flatpak/flatpak#5713]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

In git, you can always use a beginning portion of a commit's hash as shorthand, which makes copy-pasting (or typing) easier and also lets you reference it elsewhere without super long strings. However, this doesn't work in flatpak:

```
$ flatpak update --user --commit=6eedc9dfc5f042 org.gnome.gitgDevel                                                                                             

Looking for updates…
error: Server returned status 404
```

…and the 404 error is a bit cryptic. It turns out you have to use the full exact hash:

```
$ flatpak update --user --commit=6eedc9dfc5f0423dae5c69081bc8c7169bb65be96a9a82025eb1993b1a05ad9f org.gnome.gitgDevel 
Looking for updates…

        ID                            Branch          Op          Remote                 Download
 1.     org.gnome.gitgDevel           master          u           gnome-nightly          < 15.5 MB
```

It would be nice if flatpak could smartly figure out the potentially matching hash(es) from `flatpak remote-info --log the_remote the_application_id`, it has all this information. It could make an autocorrection suggestion or choice prompt if there are somehow multiple matching results.

**Comment by Henry-Hiles** [comment]
Yep, this would definitely be good. This is a big blocker for a tool I'm writing
